modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotSelfTest.java

```diff
@@ -1224,6 +1224,32 @@ public void testClusterSnapshotFinishedTryCancel() throws Exception {
         assertSnapshotCacheKeys(snpIg.cache(dfltCacheCfg.getName()));
     }
 
+    /** @throws Exception If fails. */
+    @Test
+    public void testClientHandlesSnapshotFailOnStartStage() throws Exception {
+        Ignite ignite = startGridsWithCache(2, dfltCacheCfg, CACHE_KEYS_RANGE);
+
+        Ignite cln = startClientGrid();
+
+        TestRecordingCommunicationSpi failNodeSpi = TestRecordingCommunicationSpi.spi(grid(1));
+
+        failNodeSpi.blockMessages((n, msg) -> msg instanceof SingleNodeMessage);
```

Comment:
To be sure: `failNodeSpi.blockMessages((n, msg) -> msg instanceof SingleNodeMessage && ((SingleNodeMessage)msg).type() == START_SNAPSHOT.ordinal());` ?